### ISSUE TITLE
[cfcore] SYNC PROTOCOL -- Unexpected token u in JSON at position 0 

### ISSUE THREAD

**Issue by ArjunBhuptani** [connext/indra#1296]
**Describe the bug**
The following error gets thrown when attempting to sync a proposal: 
```
2020-07-07T15:42:04.001Z E [10][CF-handleReceivedProtocolMessage] Caught error running sync protocol, aborting. Error: SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.bigNumberifyJson (/root/modules/utils/dist/json.js:10:16)
    at Function.fromJson (/root/modules/cf-core/dist/models/app-instance.js:38:496)
    at syncUntrackedProposals (/root/modules/cf-core/dist/protocol/sync.js:474:43)
    at /root/modules/cf-core/dist/protocol/sync.js:290:60
    at Generator.next (<anonymous>)
    at resume (/root/modules/cf-core/dist/protocol/sync.js:8:44)
    at /root/modules/cf-core/dist/protocol/sync.js:7:121
    at new Promise (<anonymous>)
    at Object.i.<computed> [as next] (/root/modules/cf-core/dist/protocol/sync.js:7:63)
    at ProtocolRunner.runProtocol (/root/modules/cf-core/dist/machine/protocol-runner.js:89:39)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async ProtocolRunner.runProtocolWithMessage (/root/modules/cf-core/dist/machine/protocol-runner.js:31:33)
    at async handleReceivedProtocolMessage (/root/modules/cf-core/dist/message-handling/handle-protocol-message.js:20:41)
    at async RequestHandler.callEvent (/root/modules/cf-core/dist/request-handler.js:65:13)
    at async CFCore.handleReceivedMessage (/root/modules/cf-core/dist/cfCore.js:262:13)
    at async /root/modules/cf-core/dist/cfCore.js:244:17
```

This is likely caused by proposals/commitments in `syncUntrackedProposals` being undefined, which would happen in the event that the counterparty rejected the proposal (but `numProposedApps` was still incremented).

**To Reproduce**
Not really easy to reproduce. Best bet is to run the tps bot tests and quit them halfway, then wait a minute and try to run them again.

**Comment by ArjunBhuptani** [comment]
Issue wasn't solved by the original PR, see new work here: #1303

**Comment by LayneHaber** [comment]
closing, fixed
